app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -37,34 +24,9 @@ def start_clone(clone_options)
   end
 
   def do_clone_task_check(clone_task_ref)
-    if request_type == 'clone_to_template'
-      source.with_provider_connection(:service => "PCloudTasksApi") do |api|
-        task = api.pcloud_tasks_get(phase_context[:clone_task_mor])
-        stop = (task.status != 'capturing')
-        phase_context[:cloud_api_completion_time] = Time.zone.now.utc if stop
-        return stop, task.status_detail
-      end
-    else
-      source.with_provider_connection(:service => "PCloudPVMInstancesApi") do |api|
-        instance = api.pcloud_pvminstances_get(cloud_instance_id, clone_task_ref)
-        instance_state = instance.status
-        stop = false
-
-        case instance_state
-        when 'BUILD'
-          status = 'The server is being provisioned.'
-        when 'ACTIVE'
-          stop = (instance.processors.to_f > 0) && (instance.memory.to_f > 0)
-          status = "The server has been provisioned.; #{stop ? 'Server description available.' : 'Waiting for server description.'}"
-        when 'ERROR'
-          raise MiqException::MiqProvisionError, _("An error occurred while provisioning the instance.")
-        else
-          status = "Unknown server state received from the cloud API: '#{instance_state}'"
-          _log.warn(status)
-        end
-
-        return stop, status
-      end
+    case request_type
+    when 'clone_to_template' then check_task_clone_to_template
+    else                          check_task_clone(clone_task_ref)
```

Comment:
This `case/when` is a little overkill for what could be an `if / else` or ternary operator.  Recommend the way it is done in `prepare_for_clone_task` above on line 7
```suggestion
    request_type == 'clone_to_template' ? check_task_clone_to_template : check_task_clone(clone_task_ref)
```

Reply:
Agreed, as I was thinking of more `when`'s to come, but am not sure if they will come. Force-pushed.